openerp/models.py

```diff
@@ -3419,7 +3419,10 @@ def qualify(field):
                         res2 = self._columns[f].get(cr, self._model, ids, f, user, context=context, values=result)
                         for vals in result:
                             if res2:
-                                vals[f] = res2[vals['id']]
+                                if vals['id'] in res2.keys():
+                                    vals[f] = res2[vals['id']]
+                                else:
+                                    vals[f] = []
```

Comment:
The pythonic way to avoid an error when `vals['id']` does not exist is to use `vals.get('id')`